### PR TITLE
feat(argocd-applicationset): Add ingress for webhook

### PR DESCRIPTION
**Pull request by c0deaddict**
Implements #1014

Verified working on a setup with ArgoCD with ApplicationSet controller.

Note on DCO:

If the DCO action in the integration test fails, one or more of your commits are not signed off. Please click on the *Details* link next to the DCO action for instructions on how to resolve this.

Checklist:

* [x] I have bumped the chart version according to [versioning](https://github.com/argoproj/argo-helm/blob/master/CONTRIBUTING.md#versioning)
* [x] I have updated the chart changelog with all the changes that come with this pull request according to [changelog](https://github.com/argoproj/argo-helm/blob/master/CONTRIBUTING.md#changelog).
* [x] Any new values are backwards compatible and/or have sensible default.
* [x] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/tree/master/community#contributing-to-argo).
* [x] My build is green ([troubleshooting builds](https://argoproj.github.io/argo-cd/developer-guide/ci/)).

Changes are automatically published when merged to `master`. They are not published on branches.

### CODE REVIEW

File: charts/argocd-applicationset/values.yaml [hunk 1]
```diff
@@ -146,3 +146,58 @@ extraVolumes: []
...
+    # -- Additional ingress labels
+    labels: {}
+    # -- Defines which ingress controller will implement the resource
+    ingressClassName: ""
+
+    # -- List of ingress hosts
+    ## Hostnames must be provided if Ingress is enabled.
+    ## Secrets must be manually created in the namespace
+    hosts:
+      []
```
Comment (enys) [inline charts/argocd-applicationset/values.yaml:174]: Isn't this inconsistent, having the [] on a separate line ? Even if probably more convenient

**Comment by enys** [comment]
@c0deaddict @mkilchhofer I would benefit from this one, anything I can do to help ?

File: charts/argocd-applicationset/values.yaml (hunk 1 above)
Comment (c0deaddict) [reply charts/argocd-applicationset/values.yaml]: Yes, you are right. Copied this blindly form `argo-cd/values.yaml`.

**Comment by c0deaddict** [comment]
> @c0deaddict @mkilchhofer I would benefit from this one, anything I can do to help ?

The github action "chart-test" was broken in my first attempt. I couldn't figure out why. Maybe it will work now that the update to 0.3.0 has been merged into master. If not I could really use your help to get the test to work.

**Comment by enys** [comment]
> > @c0deaddict @mkilchhofer I would benefit from this one, anything I can do to help ?
> 
> The github action "chart-test" was broken in my first attempt. I couldn't figure out why. Maybe it will work now that the update to 0.3.0 has been merged into master. If not I could really use your help to get the test to work.

It's actually linked to a few issues on the application-set controller itself, (logs and the webhook itself) requiring argocd to be setup in the same namespace. The chart test has been deactivated for the time being.

Will re-read the PR this afternoon

**Comment by enys** [comment]
Couldn't spot anything else, LGTM.
I guess it's up to @mkilchhofer

**Review by mkilchhofer — APPROVED** [review]

**Pull request merged**